### DISCUSSION TITLE
WebRTC sample broken after targeting .NET6 & latest Uno packages

### DISCUSSION THREAD

**Discussion by Eagle3386** [unoplatform/uno#8401] [Q&A] [answered]
## On which page?
https://github.com/unoplatform/Uno.Samples/tree/master/UI/WebRTC

## What's wrong?
Upgraded both projects from .NET 5 to 6, updated only Uno packages to their latest (with `nuget.org` as package repository), tried to build both projects & failed due to:
`global::Uno.Extensions.LogExtensionPoint.AmbientLoggerFactory = factory;`

That line is located at the bottom of `App.xaml.cs` but even after digging in several (sub-) repositories of Uno, I couldn't find _any_ hint about where that class ever was, nor why it got removed, nor how to fix the obvious error by "redirection" of the `factory` assignment to another property of another class as I simply can't find appropriate docs teaching about proper code for this assignment.

## Any feedback?
As a starter regarding .NET MAUI and Uno as a possible alternative due to MAUI lacking quite a lot of stuff, I find it pretty hard to get going with the platform. Especially, because using the latest VS 2022 preview (17.2 version 2.0), I received _multiple_ errors upon initial setup of the Uno solution templates although everything was set up as stated at [Get Started on Visual Studio](https://platform.uno/docs/articles/get-started-vs.html#installing-the-uno-platform-solution-templates-with-visual-studio).
Sure, I'm a software engineer & dealing with frictions as an early adopter is part of the game - but honestly, it's about 80% "undocumented, unknown error due to outdated sample/template/etc" & only 20% about the aforementioned frictions that I was mostly able to overcome in one way (replacing with proper code of the newer API) or the other (commenting broken code like the logger factory assignment).

Any help would be greatly appreciated!

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the details. This sample is still using a previous version of Uno, which unfortunately breaks with recent builds of Visual Studio and .NET. We're in the process of updating those samples so they follow the recent changes. If you want to update the sample manually in the meantime, you can [follow this guide](https://platform.uno/docs/articles/migrating-from-previous-releases.html?q=migrat#uno-40).

As a general rule though, if you're starting with a new platform, the best is to not start with preview builds of any kind, which will otherwise cause issues you may not have with stable builds.

**Reply by Eagle3386** [reply]
Thanks for the quick response & especially for the guide you've linked to, Jérôme!

While I fully agree with all that you've said, there's not much of a choice: Xamarin.Forms is nearly dead, its successor, MAUI, requires you to run a VS preview while it hasn't fully caught up with Xamarin's rich feature set & WebRTC in general - not only .NET - isn't widely tried.
Most people stick with Slack, Teams, etc., but my stakeholder requires a custom, cross-platform app...

**Reply by jeromelaban** [reply]
I agree that the ecosystem is in flux, and so you know on Uno's side, both the Xamarin targets and the .NET 6 targets will continue to evolve at the same pace feature wise for the foreseeable future.
